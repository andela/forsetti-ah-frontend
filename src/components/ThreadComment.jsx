import React from 'react';
import userIcon from '../assets/svg/user.svg';
import formatDate from '../utils/formatDate.util';

const ThreadComment = (props) => {
  const {
    comment,
    createdAt,
    commentType,
    usercomments: {
      username, firstname, image
    },
  } = props;

  return (
    <div className='card shadow-sm my-3'>
      <div className='card-body'>
        <p>{comment}</p>
        <img style={{ height: '40px' }} src={image || userIcon} alt='user-icon' />
        {' '}
        <div className='d-inline'>
          <h5 className='mt-0 d-inline'>
            {username || firstname}
            {' '}
            <br />
            <small>
              <i>
    Posted on
                {' '}
                {formatDate(createdAt)}
              </i>
              {commentType === 'criticism' ? (
                <span style={{
                  backgroundColor: '#6295bc', float: 'right', color: '#fff', padding: '8px'
                }}
                >
                criticism
                </span>
              ) : ''}
            </small>
          </h5>
        </div>
      </div>
    </div>
  );
};

export default ThreadComment;
