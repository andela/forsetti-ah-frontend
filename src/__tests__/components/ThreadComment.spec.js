import React from 'react';
import { shallow } from 'enzyme';
import ThreadComment from '../../components/ThreadComment';

const props = {
  comment: 'sample comment',
  commentType: 'criticism',
  usercomments: {
    username: 'username',
    firstname: 'firstname',
    image: 'imageurl'
  },
 };
let wrapper;

describe('<ThreadComment />', () => {
  test('should render comment component correctly', () => {
    wrapper = shallow(<ThreadComment {...props} />);
    expect(wrapper.length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
