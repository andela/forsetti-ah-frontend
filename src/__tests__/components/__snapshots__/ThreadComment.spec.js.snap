// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThreadComment /> should render comment component correctly 1`] = `ShallowWrapper {}`;
